app/packages/addons-service/provider.ts

```diff
@@ -146,37 +146,35 @@ export default class Provider {
     @task
     @waitFor
     async getUserReference() {
-        const userReference = this.store.peekRecord('user-reference', this.currentUser.user?.id);
-        if (userReference) {
-            this.userReference = userReference;
-        } else {
-            this.userReference = await this.store.findRecord('user-reference', this.currentUser.user?.id);
-        }
+        const { user } = this.currentUser;
+        const userReferences = await this.store.query('user-reference', {
+            filter: {user_uri: user?.links.iri?.toString()},
+        });
+        this.userReference = userReferences.firstObject;
     }
 
+
     @task
     @waitFor
     async getResourceReference() {
         if (this.node) {
-            const serviceNode = this.store.peekRecord('resource-reference', this.node.id);
-            if (serviceNode) {
-                this.serviceNode = serviceNode;
-            } else {
-                this.serviceNode = await this.store.findRecord('resource-reference', this.node.id);
-            }
+            const resourceRefs = await this.store.query('resource-reference', {
+                filter: {resource_uri: this.node.links.iri?.toString()},
+            });
+            this.serviceNode = resourceRefs.firstObject;
```

Comment:
Did the peekRecord not work properly?

Reply:
peekRecord uses id, but gravyvalet mints its own ids and getting the gv resource-ref for an osf resource requires looking it up with `filter[resource_uri]` (...which is less than ideal but where we've landed for now, it seems -- might be cleaner if it were fetched/cached on the `node` model?)

Reply:
I think it would still be worth doing a peekRecord here and for the userReference, but now with the correct GV ids. We instantiate this class quite a bit and taking advantage of the caching would be nice. Unless query checks for any cached records first?